README.md

```diff
@@ -16,6 +16,15 @@ gradient-free optimizers, and constrained optimization.
 * OpenBLAS or Intel MKL or LAPACK (see Armadillo site for details)
 
 
+### Installation
+
+ensmallen can be installed with CMake 3.3 or later.
+If CMake is not already available on your system, it can be obtained from https://cmake.org
```

Comment:
```suggestion
If CMake is not already available on your system, it can be obtained from https://cmake.org.
```
Sorry for the pedantry. :)

Reply:
Yeah, I left out the full stop at the end of the sentence, so the copy and paste of the address is easy.  This mainly applies to people reading the README with a text editor, or a viewer like "less".

I'm all for strictness where it's required, though allow for flexibility in corner cases like this.

Reply:
Fair point.  That missing period will bug me though. :)

Reply:
ha :) does this imply that the american date format (MM/DD/YY) bugs you as well?

Reply:
It does!  It doesn't sort right, and it also doesn't make sense.  I generally go with YYYY/MM/DD as a compromise.  At least, I think it's a compromise?  I have no idea.

Anyway I did figure out how to get my brain to rest easy on this one; let me know what you think:

```
If CMake is not already available on your system, http://cmake.org has instructions for how to download and install it.
```

I can make that change during merge.

Reply:
looks good.